### PR TITLE
Improve DeclarationExp.toChars() for inlining

### PR DESCRIPTION
**Pull request by 9rnsr**
Separated from #5292:

Do not print newline when DeclarationExp.toChars() is directly called.

### CODE REVIEW

File: src/hdrgen.d [hunk 1]
```diff
@@ -2459,7 +2454,15 @@ public:
 
     override void visit(DeclarationExp e)
     {
-        e.declaration.accept(this);
+        if (auto v = e.declaration.isVarDeclaration())
+        {
+            buf.writeByte('(');
+            visitVarDecl(v, false);
+            buf.writeByte(';');
+            buf.writeByte(')');
```
Comment (WalterBright) [inline src/hdrgen.d]: This looks strange, like it generates:
```
(int v;)
```
where would that be used?

Comment (9rnsr) [inline src/hdrgen.d]: For header generation, a `DeclarationExp` is always printed via `ExpStatement`. If `DeclarationExp.toChars()` is directly called, it occurs by our debugging (e.g.: `printf("exp = %s\n", e.toChars());`).

Yes' the print format is a little strange, but during debugging, the unfitness to regular D code is rather good to recognize `DeclarationExp`.

Comment (WalterBright) [inline src/hdrgen.d]: I read your comment several times and do not understand it. Are you saying that the code currently generates syntactically incorrect code? or not? or you fixed it so it does?

By the way, thanks for splitting up the PRs into digestible chunks. It makes things a lot easier.

Comment (9rnsr) [inline src/hdrgen.d]: This is a change for debugging. It's not a bugfix.

This PR also changes `visit(ExpStatement s)` function. After that change, `visit(DeclarationExp e)` will never be reachable from any normal dmd execution. So, this strange output will be shown only when we'll directly use `DeclarationExp.toChars()` for debugging.

While inliner debugging, I repeatedly prints the expanded expression tree. For that, the modified output format for `DeclarationExp` will help me.

Comment (WalterBright) [inline src/hdrgen.d]: Please add comments to the code describing its purpose - because it just looks wrong and does emit incorrect D code.

Comment (9rnsr) [inline src/hdrgen.d]: OK, I added comments to describe the purpose.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**